tasks/tasks.go

```diff
@@ -5,16 +5,18 @@ import (
 	"math/rand"
 	"sync"
 	"time"
+
+	log "github.com/Sirupsen/logrus"
 )
 
 // Status is the status of the task.
 type Status int
 
 // TaskExecuter executes a series of task functions.
 type TaskExecuter interface {
-	QueueTask(task func() error) string
+	QueueTask(task func(string) error) string
```

Comment:
It's the task id. I needed a way for the task itself to be aware of its id (for logging). This isn't _that_ ideal but I don't know a better way to do it.